scripts/functional-tests.sh

```diff
@@ -67,6 +67,9 @@ function init_test_script() {
     HOME="${TEST_DIR}"  # to ensure platformdirs uses the test directory for the duration of this script
     printf 'changing home directory to "%s", was: "%s"\n' "${HOME}" "${prev_home}"
 
+    # DO NOT REMOVE THE VARIABLE ASSIGNMENTS BELOW
```

Comment:
Sorry but: Which variables are we talking about here?.. There are several sections below. Does the comment apply to all of them?

Reply:
Yes

Reply:
OK. But isn't it *obvious* that they are needed? (They are used elsewhere in the script.) And how is `ilab` creating directories relevant here? I don't think `ilab` itself cares about any of the variables listed below.

I don't understand what this comment is trying to address / explain. Can you elaborate?

Reply:
@booxter further down in the test script we are creating the directories for all of these variables. For those familiar with the init process, this may seem redundant as `ilab` has a function called `ensure_storage_directories_exist` where it recursively creates the storage directories that will be used in subsequent calls. But for the purpose of knowing how ilab works, we expect that these directories will already exist. So `ilab` is only responsible for those that it must create.

As I'm writing this out though, I'm realizing that we should probably also handle the case where these directories don't exist and `ilab` creates them manually, as this may be more realistic for some of our users.